python/pdstools/ih/Plots.py

```diff
@@ -8,4 +14,130 @@
+        positive_labels: Optional[List[str]] = None,
+        negative_labels: Optional[List[str]] = None,
+        reference_values: Optional[Dict] = None,
+        title: Optional[str] = "Experiment Overview",
+        return_df:Optional[bool] = False,
+    ):
+        # TODO currently only supporting a single by
+
+        plot_data = self.ih.aggregates.summary_by_experiment(
+            experiment_field=experiment_field,
+            by=by,
+            positive_labels=positive_labels,
+            negative_labels=negative_labels,
+        )
+
+        if return_df:
+            return plot_data
+        
+        plot_data = plot_data.collect()
+
+        cols = plot_data[by].unique().shape[0]
+        rows = plot_data[experiment_field].unique().shape[0]
+
+        fig = make_subplots(
+            rows=rows,
+            cols=cols,
+            specs=[[{"type": "indicator"} for c in range(cols)] for t in range(rows)],
+        )
+        fig.update_layout(
+            height=270 * rows,
+            autosize=True,
+            title=title,
+            margin=dict(b=10, t=120, l=10, r=10),
+        )
+        index = 0
+        for row in plot_data.iter_rows(named=True):
+            ref_value = (
+                reference_values.get(row[by], None) if reference_values else None
+            )
+            gauge = {
+                "axis": {"tickformat": ",.2%"},
+                "threshold": {
+                    "line": {"color": "red", "width": 2},
+                    "thickness": 0.75,
+                    "value": ref_value,
+                },
+            }
+            if ref_value:
+                if row["CTR"] < ref_value:
+                    gauge = {
+                        "axis": {"tickformat": ",.2%"},
+                        "bar": {
+                            "color": (
+                                "#EC5300"
+                                if row["CTR"] < (0.75 * ref_value)
+                                else "#EC9B00"
+                            )
+                        },
+                        "threshold": {
+                            "line": {"color": "red", "width": 2},
+                            "thickness": 0.75,
+                            "value": ref_value,
+                        },
+                    }
+
+            trace1 = go.Indicator(
+                mode="gauge+number+delta",
+                number={"valueformat": ",.2%"},
+                value=row["CTR"],
+                delta={"reference": ref_value, "valueformat": ",.2%"},
+                title={"text": f"{row[by]}: {row[experiment_field]}"},
+                gauge=gauge,
+            )
+            r, c = divmod(index, cols)
+            fig.add_trace(trace1, row=(r + 1), col=(c + 1))
+            index = index + 1
+
+        return fig
+
+    def tree_map(
+        self,
+        experiment_field: str,
+        by: Optional[List[str]] = None,
+        positive_labels: List[str] = None,
```

Comment:
I think this should be type hinted as `Optional[List[str]]` too - None is not a list type